### PR TITLE
Remove useless code

### PR DESCRIPTION
**Pull request by mlevieux**
Unnecessary conversions and such were removed

### CODE REVIEW

File: app_secret.go [hunk 1]
```diff
@@ -11,8 +11,9 @@ const appCreationNature = 1
 
 func newAppId(appSecret []byte) []byte {
 	publicKey := appSecret[AppSecretSize-AppPublicKeySize : AppSecretSize]
-	author := make([]byte, 32)
-	payload := append([]byte{appCreationNature}, author...)
+
+	payload := make([]byte, 33)
+	payload[0] = appCreationNature
```
Comment (blastrock) [inline app_secret.go:16]: The previous style was better imho. I'd even make it more verbose

```go
nature := []byte{appCreationNature}
author := make([]byte, 32)
payload := make([]byte, 0)
payload := append(payload, nature..., author..., publicKey...) // can't remember if this is legal
```

The idea is that what's written here should not be too hard to correlate with [the spec](https://github.com/TankerHQ/spec/blob/master/blocks_format.md#block-format)

File: identity.go [hunk 2]
```diff
@@ -135,22 +134,17 @@ func UpgradeIdentity(b64Identity string) (*string, error) {
 
 
 func generateIdentity(config config, userIDString string) (*identity, error) {
-	generatedAppID := newAppId(config.AppSecret)
-
-	if !bytes.Equal(generatedAppID, config.AppID) {
+	if !bytes.Equal(newAppId(config.AppSecret), config.AppID) {
```
Comment (blastrock) [inline identity.go:137]: Related to the other MR. Now that I see this, I think newAppId might not be a good name. It's more like getAppId(appSecret). `newAppId` suggests that you'll get a different one on each call, no?

**Review by blastrock — APPROVED** [review]

File: app_secret.go (hunk 1 above)
Comment (JMounier) [reply app_secret.go:16]: More over, the actual code should not work.
```go
payload := make([]byte, 33)
payload = append(payload, publicKey...)
if len(payload) != 33 {
  assert("publicKey was appended after the 32 zeros");
} 
```
We should use `make([]byte, 0, 33)` and then the same `append()` as before

File: identity.go (hunk 2 above)
Comment (mlevieux) [reply identity.go:137]: Indeed the name is misleading, will change this!

File: app_secret.go (hunk 1 above)
Comment (mlevieux) [reply app_secret.go:16]: @blastrock unfortunately what you suggest is not legal and requires chained appends, like `append(payload, append(nature, append(author, publicKey...)...)...)`
What I tried to do here is avoid the unnecessary allocation of author that's clearly only a sequence of zeros.

Following my previous note, @JMounier I'm not sure I get what you mean, the previous code had a final state of:
```
[appCreationNature, 0, 0, 0... (32 zeros in total)]
```
which is exactly what this does -> allocate a 33 bytes slice and change only the first one? Correct me if I'm wrong!

Comment (JMounier) [reply app_secret.go:16]: My bad. It felt wrong to pad with 32 zeros for no obvious reason. It seems to be a placeholder for an author

**Review by JMounier — APPROVED** [review]

File: app_secret.go (hunk 1 above)
Comment (mlevieux) [reply app_secret.go:16]: I remade a change that I think better conveys the idea of the spec while avoiding the unnecessary allocation, WDYT @blastrock ?

Comment (blastrock) [reply app_secret.go:16]: that's better :)

**Review by blastrock — APPROVED** [review]

**Pull request merged**